src/pages/index.js

```diff
@@ -98,32 +178,38 @@ function handleImageClick(link, name) {
 
 // refactored renderCard function using the Card class
 function renderCard(item, method = "prepend") {
-  const card = new Card(item, "#card-template", handleImageClick);
+  const card = new Card(
+    item,
+    "#card-template",
+    handleImageClick,
+    deleteCardPopup,
+    handleDeleteCard,
+    api
+  );
   const cardElement = card.getView();
+  cardElement.dataset.id = item._id; // store the cardId to the card element
   cardListElement[method](cardElement);
```

Comment:
Only `Section` should add any card to the DOM.

It will be like this:

```js

function renderCard(item, method = "addItem") {
  const card = new Card(
    item,
    "#card-template",
    handleImageClick,
    deleteCardPopup,
    handleDeleteCard,
    api
  );
  const cardElement = card.getView();
  cardElement.dataset.id = item._id;  
  section[method](cardElement);
}
```